### PR TITLE
Unit test for CollectionUtils

### PR DESCRIPTION
**Pull request by 4D17Y4**
## Purpose / Description
Added CollectionUtilsTest.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Test
+    public void testGetLastListElement() {
+        assertThat(CollectionUtils.getLastListElement(testList), Matchers.is(3));
+    }
+
+    @Test
+    public void testAddAll() {
+        List<Integer> toTest = new ArrayList<>();
+        CollectionUtils.addAll(toTest, testList);
+        assertThat("List should be same", toTest.equals(testList));
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.java:44]: There should be assertion dealing with list specifically. Can you please check.

They return a detailled explanation of the difference so are better

**Review by david-allison-1 — APPROVED** [review]
Looks good!

Useful but optional, let me know:

* `getLastLastElement` on an empty list - it's defined to throw
* testAddAll - ensure the order of added elements is the same
* testAddAll - ensure count is changed

File: AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.java (hunk 1 above)
Comment (4D17Y4) [reply AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.java]: Done,
Thanks.

**Comment by 4D17Y4** [comment]
> * `getLastLastElement` on an empty list - it's defined to throw

Done.

> * testAddAll - ensure the order of added elements is the same
> * testAddAll - ensure count is changed

As I've used equalsArrayList now, I think these are not required.

File: AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        add(2);
+        add(3);
+    }};
+
+    @Test
+    public void testGetLastListElement() {
+        assertThat(CollectionUtils.getLastListElement(testList), is(3));
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/utils/CollectionUtilsTest.java:42]: nice way to test a boundary conditions - this is a good Java code quality trick right here!
